utils/python/CIME/XML/entry_id.py

```diff
@@ -151,36 +182,71 @@ def get_elements_from_child_content(self, childname, childcontent):
         nodes = self.get_nodes("entry")
         elements = []
         for node in nodes:
-            childnode = self.get_node(childname,root=node)
+            childnode = self.get_optional_node(childname,root=node)
+            expect(childnode is not None,"No childname %s for id %s"%(childname,node.get("id")))
             content = childnode.text
             if content == childcontent:
-                elements.append(node)
+                elements.append(deepcopy(node))
```

Comment:
But that is exactly the intent.  We need a copy that does not modify the original because we are in the process of creating a new object both in memory and on file.